### PR TITLE
don't created related submissions when cloning (WP-647)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

**Review by dimitrystd — APPROVED** [review]

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 1]
```diff
@@ -193,7 +198,11 @@ public function clone(CloneRequest $request): void
                     if ($title !== '') {
                         $submission->setSourceTitle($title);
                     }
-                    $submission->getFileUri();
+                    try {
+                        $submission->getFileUri();
+                    } catch (SmartlingInvalidFactoryArgumentException $e) {
+                        $this->getLogger()->error(sprintf('Failed to set submission file uri while cloning: %s, sourceBlogId=%s, sourceId=%s', $e->getMessage(), $sourceBlogId, $submissionArray[SubmissionEntity::FIELD_SOURCE_ID]));
```
Comment (dimitrystd) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:204]: Why does the message says "set" when you actually read filename?

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 2]
```diff
@@ -181,7 +181,12 @@ public function clone(CloneRequest $request): void
             foreach ($sources as $source) {
                 $submissionArray[SubmissionEntity::FIELD_CONTENT_TYPE] = $source['type'];
                 $submissionArray[SubmissionEntity::FIELD_SOURCE_ID] = (int)$source['id'];
-                $existing = ArrayHelper::first($this->submissionManager->find($submissionArray));
```
Comment (dimitrystd) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:184]: Do I understand correctly that before we searched by ALL fields and now only by 4?

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:204]: This function is incorrectly named, I've previously tried to comment: https://github.com/Smartling/wordpress-localization-plugin/commit/da985983e15b3e5f92f121c4dab7e13f282246c5#diff-734781b35b45fb97be3758e309ef129b85ca551d4376e945f1c5f69b87687fe5R556

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:184]: Yes

**Pull request merged**